File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4ruby/lib/Excepciones/direccion.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Direccion
  attr_reader :calle, :numero, :codigoPostal
  attr_accessor :ciudad
  def initialize(calle, numero, cp, ciudad)
    @calle = calle
    @numero = numero
    @codigoPostal = cp
    @ciudad = ciudad    
  end
  # Método en el que se lanza una excepción
  def codigoPostal(cp)
    if (cp.size <=> 5)  
      raise "código postal erroneo"
    end
    @codigoPostal = cp
  end
 
end

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4java/src/ConstructorCopiaProfunda/Direccion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ConstructorCopiaProfunda;

/**
 *
 * @author ana
 */
class Direccion {
     String calle;
      int numero;
      String codigoPostal;
      String ciudad;

    public Direccion(Direccion dc) {
        this.calle = dc.calle;
        this.numero =dc.numero;
        this.codigoPostal = dc.codigoPostal;
        this.ciudad = dc.ciudad;    
    } 
     public Direccion(String calle, int num, String cp, String ciudad) {
        this.calle = calle;
        this.numero = num;
        this.codigoPostal = cp;
        this.ciudad = ciudad;    
    } 
}
